dolphin/drivers/manager.py

```diff
@@ -11,18 +11,31 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import threading
 
 from oslo_log import log
 
 
 LOG = log.getLogger(__name__)
+lock = threading.Lock()
 
 DRIVER_MAPPING = {
     "fake_storage": "dolphin.drivers.fake_storage.FakeStorageDriver"
 }
 
 
-class DriverManager(object):
+class Singleton(type):
+    _instances = {}
```

Comment:
Why do we need a list of instances? Please put the reason as comment

Reply:
Use dict here inorder to every class which want singleton pattern can inherit this class to implement singleton pattern.
The "key" and "value" type of this dict is "class" and "instance"